source/common/upstream/edf_scheduler.h

```diff
@@ -0,0 +1,97 @@
+#include <cstdint>
+#include <queue>
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Upstream {
+
+// It's not sufficient to use trace level logging, since it becomes far too noisy for a number of
+// tests, so we can kill EDF trace debug here.
+#define EDF_DEBUG 0
+
+#if EDF_DEBUG
+#define EDF_TRACE(fmt...) ENVOY_LOG_MISC(trace, fmt)
+#else
+#define EDF_TRACE(fmt...)
+#endif
+
+// EDF scheduler. Entries have deadlines set at current time + 1 / weight, providing weighted round
```

Comment:
Can we spell out EDF and link to some docs (Wikipedia or otherwise) for the casual reader?